File: Data Structure/Array or Matrix/Medium/LET_no565_Array Nesting.cpp
class Solution {
public:
    int arrayNesting(vector<int>& nums) {
        int res = 0;
        for (int i = 0; i < nums.size(); i++) {
            int cur = 0;
            for (int j = i; nums[j] >= 0; cur++) {
                int temp = nums[j];
                nums[j] = -1;
                j = temp;
            }
            res = max(res, cur);
        }
        return res;
    }
};

File: Data Structure/Array or Matrix/Medium/LET_no565_Array Nesting.py
class Solution:
    def arrayNesting(self, nums: List[int]) -> int:
        res = 0
        for i in range(len(nums)):
            cur = 0
            j = i
            while nums[j] >= 0:
                cur += 1
                temp = nums[j]
                nums[j] = -1
                j = temp
            res = max(res, cur)
        return res